### PR TITLE
Extract common code from DiscretizedHEX models into a partial model. 

### PR DESCRIPTION
**Pull request by CamielCartignij**
Simplified both the discretized and NTU heat exchangers to be derived from a partial class. 
- Added the summary record to the NTU heat exchangers 
- Changed the flow direction through the `ThermalConductor` element for crossflow heat exchangers to be consistent with counterflow exchangers.

### CODE REVIEW

File: ThermofluidStream/HeatExchangers/Internal/PartialDiscretizedHEX.mo [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+public
+  Modelica.Thermal.HeatTransfer.Components.ThermalConductor thermalConductor[nCells](each G=G/nCells) annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=90,
+        origin={0,0})));
+
+  ConductionElementB thermalElementB[nCells] annotation (Placement(transformation(extent={{-10,90},{10,70}})));
+  ConductionElementA thermalElementA[nCells] annotation (Placement(transformation(extent={{10,-90},{-10,-70}})));
+
+  Interfaces.Inlet inletB(redeclare package Medium = MediumB) annotation (Placement(transformation(extent={{-110,70},{-90,90}}), iconTransformation(extent=if (nCellsParallel == 1) then {{-110,70},{-90,90}} else {{110,-90},{90,-70}})));
```
Comment (CamielCartignij) [inline ThermofluidStream/HeatExchangers/Internal/PartialDiscretizedHEX.mo:65]: If there are any neater methods to have the icon connectors in different positions for counterflow vs crossflow this should still still be implemented.

Comment (tobolar) [reply ThermofluidStream/HeatExchangers/Internal/PartialDiscretizedHEX.mo:65]: Nothing that I know so far. 

But the side effect of your implemention is that the icon connectors of cross elements change their position for `nCells=1`, which is not the case for the original implementation of `DiscretizedCrossFlowHEX` and `DiscretizedCrossFlowHEX_FR`. Is this still the desired behavior? Optionally, a new protected parameter could be defined just for this.

**Review by tobolar — CHANGES_REQUESTED** [review]
The documentation shall be improved:
- `TFS.HeatExchangers.Internal.PartialNTU` is only copy/paste of the documentation of `CounterFlowNTU`,
- `TFS.HeatExchangers.Internal.PartialDiscretizedHEX` is missing competely.

(And, I've added some inline comments.)

---

Using base class `PartialDiscretizedHEX` in `TFS.HeatExchangers.DiscretizedCrossFlowHEX` and `TFS.HeatExchangers.DiscretizedCrossFlowHEX_FR`, the position of both `thermalElementA` and `thermalElementB`changes, but not the orientation of `thermalConductor`. 

To reach a nice diagram, the ports of `thermalConductor` now connect different thermal elements, which leads to sign change in `thermalConductor.Q_flow` and `dT`. 
(Consider i.g. `condenser.thermalConductor[1].Q_flow` in example `TFS.HeatExchangers.Tests.Condenser`.)

This could potentially lead to problems when a user uses these variables (for whatever reason).
@nieweber What do you mean? Is this a problem or can we simply accept this?

File: ThermofluidStream/HeatExchangers/Internal/PartialDiscretizedHEX.mo (hunk 1 above)
Comment (CamielCartignij) [reply ThermofluidStream/HeatExchangers/Internal/PartialDiscretizedHEX.mo:65]: You are right, I will correct it to work the same as the NTU heat exchangers

File: ThermofluidStream/HeatExchangers/Internal/PartialDiscretizedHEX.mo [hunk 2]
```diff
@@ -62,10 +63,10 @@ public
...
+  Interfaces.Inlet inletB(redeclare package Medium = MediumB) annotation (Placement(transformation(extent={{-110,70},{-90,90}}), iconTransformation(extent=if not crossFlow then {{-110,70},{-90,90}} else {{110,-90},{90,-70}})));
+  Interfaces.Outlet outletB(redeclare package Medium = MediumB) annotation (Placement(transformation(extent={{90,70},{110,90}}), iconTransformation(extent=if not crossFlow then {{90,70},{110,90}} else {{-90,-90},{-110,-70}})));
+  Interfaces.Inlet inletA(redeclare package Medium = MediumA) annotation (Placement(transformation(extent={{110,-90},{90,-70}}), iconTransformation(extent=if not crossFlow then {{110,-90},{90,-70}} else {{-110,-10},{-90,10}}, rotation=if not crossFlow then 0 else -90)));
+  Interfaces.Outlet outletA(redeclare package Medium = MediumA) annotation (Placement(transformation(extent={{-90,-90},{-110,-70}}), iconTransformation(extent=if not crossFlow then {{-90,-90},{-110,-70}} else {{90,-10},{110,10}}, rotation=if not crossFlow then 0 else -90)));
```
Comment (tobolar) [inline ThermofluidStream/HeatExchangers/Internal/PartialDiscretizedHEX.mo:69]: Prefer `if crossFlow ..` over `if not crossClow...` for better readability.

File: ThermofluidStream/HeatExchangers/Internal/package.order [hunk 3]
```diff
@@ -4,3 +4,5 @@ ConductionElementHEX_twoPhase
 DiscretizedHEXSummary
 DiscretizedHexIcon
 calculateEfficiency
+PartialDiscretizedHEX
+PartialNTU
```
[the remark this answers is not in the document]
Comment (CamielCartignij) [reply ThermofluidStream/HeatExchangers/Internal/package.order:8]: Currently I focused on being consistent within the TFS library. Is there a desire to convert all partial classes to a separate BaseClasses subpackage? If so I can update. (@nieweber)

[the remark this answers is not in the document]
Comment (tobolar) [reply ThermofluidStream/HeatExchangers/Internal/package.order:8]: This is fine for now. Just for your information.

**Comment by tobolar** [comment]
> There are 2 options here;

Well, there some more options:

3. Rotate `thermalConductor` by -90 degree to enable connections sideways.
4. `thermalConductor[nCells](each G= (if crossFlow then -1 else 1) * G/nCells` - but this would lead to negative G and is really awful!

Since `thermalConductor` is public, IMO we shall care about not braking backward compatibility.

**Comment by nieweber** [comment]
> The documentation shall be improved:
> 
> * `TFS.HeatExchangers.Internal.PartialNTU` is only copy/paste of the documentation of `CounterFlowNTU`,
> * `TFS.HeatExchangers.Internal.PartialDiscretizedHEX` is missing competely.
> 
> (And, I've added some inline comments.)
> 
> Using base class `PartialDiscretizedHEX` in `TFS.HeatExchangers.DiscretizedCrossFlowHEX` and `TFS.HeatExchangers.DiscretizedCrossFlowHEX_FR`, the position of both `thermalElementA` and `thermalElementB`changes, bot not the orientation of `thermalConductor`.
> 
> To reach a nice diagram, the ports of `thermalConductor` now connect different thermal elements, which leads to sign change in `thermalConductor.Q_flow` and `dT`. (Consider i.g. `condenser.thermalConductor[1].Q_flow` in example `TFS.HeatExchangers.Tests.Condenser`.)
> 
> This could potentially lead to problems when a user uses these variables (for whatever reason). @nieweber What do you mean? Is this a problem or can we simply accept this?

In general I would suggest that it is reasonable to connect side A of the HEX (`thermalElementA`) with port_a of the `thermalConductor`. In the `HeatExchangers.Tests.FlowRes` this leads to a negative `dT` in the `thermalConductor` and this should be correct. As the sign change is only on the `thermalConductor`-level and not on the HX level, there are hopefully not that many users affected. Anyways we will document it in the Release Notes to make sure everybody is in formed and aware of the sign change.

**Comment by tobolar** [comment]
> Anyways we will document it in the Release Notes to make sure everybody is in formed and aware of the sign change.

@CamielCartignij Please add this to release notes (`TFS.UsersGuide.ReleaseNotes`).

**Review by tobolar — APPROVED** [review]
Fine to me.

**Pull request merged**